confidential-datahub/kms/src/plugins/aliyun/client.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::BTreeMap;
+
+use async_trait::async_trait;
+use base64::engine::general_purpose::STANDARD;
+use base64::Engine;
+use chrono::Utc;
+use log::error;
+use prost::Message;
+use reqwest::Certificate;
+use reqwest::{header::HeaderMap, ClientBuilder};
+use serde_json::Value;
+use sha2::{Digest, Sha256};
+use tokio::fs;
+
+use crate::plugins::aliyun::client::dkms_api::{DecryptRequest, EncryptRequest};
+use crate::{Decryptor, Encryptor, OperationContext, ProviderSetting};
+use crate::{Error, Result};
+
+use super::annotations::{CipherContext, KmsContext};
+use super::credential::Credential;
+
+pub mod dkms_api {
+    tonic::include_proto!("dkms_api");
+}
+
+pub struct AliyunKmsClient {
+    http_client: reqwest::Client,
+    credential: Credential,
+    kms_instance_id: String,
+    endpoint: String,
+}
+
+const IN_POD_DEFAULT_KEY_PATH: &str = "/run/kms-credential/aliyun";
+
+impl AliyunKmsClient {
+    fn read_kms_instance_cert(cert_pem: &[u8]) -> Result<Certificate> {
+        let kms_instance_ca_cert = Certificate::from_pem(cert_pem)
+            .map_err(|e| Error::AliyunKmsError(format!("read kms instance ca cert failed: {e}")))?;
+        Ok(kms_instance_ca_cert)
+    }
+
+    pub fn new(
+        client_key: &str,
+        kms_instance_id: &str,
+        pswd: &str,
```

Comment:
nit: `pswd` -> `pwd`

Reply:
ok, it is actually password, let me rename it then